container-type does not force layout containment, but does force an independent formatting context

karlcow opened the pull request:
#### 8e37e86df7eba152d1f8ad18012f237a5d2b2884
<pre>
container-type does not force layout containment, but does force an independent formatting context
<a href="https://bugs.webkit.org/show_bug.cgi?id=277122">https://bugs.webkit.org/show_bug.cgi?id=277122</a>
<a href="https://rdar.apple.com/132549134">rdar://132549134</a>

Reviewed by NOBODY (OOPS!).

container-type does not force layout containment, but does force
an independent formatting context by CSS WG resolution on
<a href="https://github.com/w3c/csswg-drafts/issues/10544">https://github.com/w3c/csswg-drafts/issues/10544</a>

This takes into account the latest import for container queries
WPT tests formatting
<a href="https://github.com/WebKit/WebKit/pull/31273">https://github.com/WebKit/WebKit/pull/31273</a>
<a href="https://github.com/web-platform-tests/wpt/commit/27c89c6f06af177cbd8f2ba5517be3c110fa50a0">https://github.com/web-platform-tests/wpt/commit/27c89c6f06af177cbd8f2ba5517be3c110fa50a0</a>

* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::establishesIndependentFormattingContext const):
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
(WebCore::StyleRareNonInheritedData::usedContain const):
* LayoutTests/imported/w3c/web-platform-tests/css/css-conditional/container-queries/*
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e37e86df7eba152d1f8ad18012f237a5d2b2884

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60065 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39413 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12617 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/63983 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10595 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62195 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47085 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/10809 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48662 "Found 26 new test failures: fast/css/container-query-style-sharing.html http/tests/media/track/track-webvtt-vertical-multi-line.html imported/w3c/web-platform-tests/css/css-cascade/scope-container.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/canvas-as-container-003.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/canvas-as-container-004.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/change-display-in-container.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-gradient-invalidation.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-gradient.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-rule-cache.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-sharing-via-rule-node.html ... (failure)") | [❌ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7396 "Found 1 new test failure: fast/css/container-query-style-sharing.html (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62096 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36749 "Found 4 new test failures: fast/css/container-query-style-sharing.html media/track/track-webvtt-no-snap-to-lines-overlap.html media/track/track-webvtt-snap-to-lines-inline-style.html media/track/track-webvtt-snap-to-lines-left-right.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52027 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29504 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33454 "Found 26 new test failures: imported/w3c/web-platform-tests/css/css-cascade/scope-container.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/canvas-as-container-003.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/canvas-as-container-004.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/change-display-in-container.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-gradient-invalidation.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-gradient.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-rule-cache.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-sharing-via-rule-node.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/counters-in-container-dynamic.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/counters-in-container.html ... (failure)") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9266 "Found 60 new test failures: fast/css/container-query-style-sharing.html fast/repaint/inline-box-with-self-paint-layer.html fonts/font-weight-invalid-crash.html fonts/ligature.html fullscreen/full-screen-document-background-color.html http/tests/media/track/track-webvtt-vertical-multi-line.html http/tests/performance/performance-resource-timing-cross-origin-media-with-cors.html http/tests/security/contentSecurityPolicy/block-all-mixed-content/insecure-script-in-iframe.html http/tests/xmlhttprequest/basic-auth.html imported/w3c/web-platform-tests/content-security-policy/reporting-api/reporting-api-report-only-sends-reports-on-violation.https.sub.html ... (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9515 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55371 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9544 "Found 33 new test failures: fast/css/container-query-style-sharing.html http/tests/media/track/track-webvtt-vertical-multi-line.html imported/w3c/web-platform-tests/css/css-cascade/scope-container.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/canvas-as-container-003.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/canvas-as-container-004.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/change-display-in-container.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-gradient-invalidation.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-gradient.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-rule-cache.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-sharing-via-rule-node.html ... (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65715 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/3995 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9411 "Found 33 new test failures: fast/css/container-query-style-sharing.html http/tests/media/media-element-frame-destroyed-crash.html http/tests/media/track/track-webvtt-vertical-multi-line.html imported/w3c/web-platform-tests/css/css-cascade/scope-container.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/canvas-as-container-003.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/canvas-as-container-004.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/change-display-in-container.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-gradient-invalidation.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-gradient.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-rule-cache.html ... (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56016 "Found 28 new test failures: fast/css/container-query-style-sharing.html http/tests/media/track/track-webvtt-vertical-multi-line.html imported/w3c/web-platform-tests/css/css-cascade/scope-container.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/canvas-as-container-003.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/canvas-as-container-004.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/change-display-in-container.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-gradient-invalidation.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-gradient.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-rule-cache.html imported/w3c/web-platform-tests/css/css-conditional/container-queries/container-units-sharing-via-rule-node.html ... (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4013 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52010 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56167 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3320 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35226 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36308 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37396 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/36052 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/rendering/RenderElement.cpp:
```diff
@@ -2407,7 +2407,7 @@ bool RenderElement::createsNewFormattingContext() const
 bool RenderElement::establishesIndependentFormattingContext() const
 {
-    return isFloatingOrOutOfFlowPositioned() || (isBlockBox() && hasPotentiallyScrollableOverflow()) || style().containsLayout() || paintContainmentApplies() || (style().isDisplayBlockLevel() && style().blockStepSize());
+    return containerType() != ContainerType::Normal || isFloatingOrOutOfFlowPositioned() || (isBlockBox() && hasPotentiallyScrollableOverflow()) || style().containsLayout() || paintContainmentApplies() || (style().isDisplayBlockLevel() && style().blockStepSize());
```
```suggestion
    return style().containerType() != ContainerType::Normal || isFloatingOrOutOfFlowPositioned() || (isBlockBox() && hasPotentiallyScrollableOverflow()) || style().containsLayout() || paintContainmentApplies() || (style().isDisplayBlockLevel() && style().blockStepSize());
```

It should be `style().containerType()`

karlcow committed:
container-type does not force layout containment, but does force an independent formatting context
https://bugs.webkit.org/show_bug.cgi?id=277122
rdar://132549134

Reviewed by NOBODY (OOPS!).

container-type does not force layout containment, but does force
an independent formatting context by CSS WG resolution on
https://github.com/w3c/csswg-drafts/issues/10544

This takes into account the latest import for container queries
WPT tests formatting
https://github.com/WebKit/WebKit/pull/31273
https://github.com/web-platform-tests/wpt/commit/27c89c6f06af177cbd8f2ba5517be3c110fa50a0

* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::establishesIndependentFormattingContext const):
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
(WebCore::StyleRareNonInheritedData::usedContain const):
* LayoutTests/imported/w3c/web-platform-tests/css/css-conditional/container-queries/*
